From 06a6d4306a09d7d69d9aed1a793aff172a2aa0b3 Mon Sep 17 00:00:00 2001
From: Evan Huus <eapache@gmail.com>
Date: Sun, 1 Mar 2015 18:45:24 -0500
Subject: [PATCH 4/6] scsi-osd: fix overflow leading to infinite loop

Thanks to Vlad Tsyrklevich for the report.

Bug: 11024
Change-Id: I9defc6fc72339b2f6ba12b46172b776d798647cd
Reviewed-on: https://code.wireshark.org/review/7463
Petri-Dish: Evan Huus <eapache@gmail.com>
Tested-by: Petri Dish Buildbot <buildbot-no-reply@wireshark.org>
Reviewed-by: Evan Huus <eapache@gmail.com>
(cherry picked from commit c35ca6c051adb28c321db54cc138f18637977c9a)
Reviewed-on: https://code.wireshark.org/review/7485
Reviewed-on: https://code.wireshark.org/review/7707
Reviewed-by: Balint Reczey <balint@balintreczey.hu>
Tested-by: Balint Reczey <balint@balintreczey.hu>
---
 epan/dissectors/packet-scsi-osd.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/epan/dissectors/packet-scsi-osd.c b/epan/dissectors/packet-scsi-osd.c
index 4aad143..deb00f0 100644
--- a/epan/dissectors/packet-scsi-osd.c
+++ b/epan/dissectors/packet-scsi-osd.c
@@ -1021,7 +1021,10 @@ dissect_osd2_cdb_continuation(packet_info *pinfo, tvbuff_t *tvb, guint32 offset,
             expert_add_info(pinfo, item_length, &ei_osd2_cdb_continuation_descriptor_length_invalid);
             return;
         }
-        offset += length+padlen;
+        /* check for overflow */
+        if (offset + length + padlen > offset) {
+            offset += length+padlen;
+        }
     }
 
 }
-- 
2.1.4

